src/Passes/docs/src/DeveloperGuide/WritingComponent.md

```diff
@@ -0,0 +1,190 @@
+                          "Demonstrating how configuration works.");
+    config.addParameter(message_, "message",
+                        "Message which is printed when setting the component up.");
+  }
+
+  // ...
+};
+```
+
+The purpose of the `setup(config)` function is to inform the configuration manager about what the name of the configuration section and its description is as well as defining all settings and bind them to C++ variables. The benefit of this approach is that all configuration parameters for the component will be available immediately after the component is loaded by the tool.
+
+The final code to manage the configuration reads:
+
+```c++
+class HelloWorldConfig
+{
+public:
+  using String = std::string;
+
+  void setup(ConfigurationManager &config)
+  {
+    config.setSectionName("Hello world configuration",
+                          "Demonstrating how configuration works.");
+    config.addParameter(message_, "message",
+                        "Message which is printed when setting the component up.");
+  }
+
+  String message() const
+  {
+    return message_;
+  }
+
+private:
+  String message_{"Hello world"};
+};
+```
+
+With the configuration in place, the next thing we concern ourselves with is loading the component. This is the functionality that registers the configuration together with an ID and a profile setup function. In our case, the setup function should just print a message given a `HelloWorldConfig` instance. The corresponding component registration reads:
+
+```c++
+extern "C" void loadComponent(IProfileGenerator *generator)
+{
+  generator->registerProfileComponent<HelloWorldConfig>(
+      "hello-world",
+      [](HelloWorldConfig const &cfg, IProfileGenerator * /*generator*/, Profile & /*profile*/) {
+        std::cout << "Message: " << cfg.message() << std::endl;
+      });
+}
+```
+
+In this example, we will only concern ourselves with how to use the configuration and we will ignore `generator` and `profile` for now. The full source code to this example can be found in `ComponentExamples/HelloWorld` and it can be compiled through following steps (startig from the Passes root folder):
+
+```sh
+mkdir Debug
+cd Debug
+cmake ..
+make HelloWorld
+```
+
+This will generate a `HelloWorld` dynamic library with path `./ComponentExamples/libHelloWorld.(dylib|so|dll)`.
+
+## Loading the component
+
+```sh
+% ./Source/Apps/qat --load ./ComponentExamples/libHelloWorld.dylib
+Usage: ./Source/Apps/qat [options] filename
+
+...
+
+Hello world configuration - Demonstration configuration for building a component boilerplate.
+
+--message                                         Message which is printed when setting the component up. Default: Hello world
+
+
+...
+```
+
+Next we generate a QIR to test this with:
+
+```
+pushd ../QirExamples/LoopRecursion/QSharpVersion
+make
+popd
+```
+
+To test that the setup function is invoked upon generating the profile, we run
+
```

Comment:
"generating the profile"?
Did you mean "loading the profile" or "generating the specialised QIRs" or other?

Reply:
I've attempted to clarify this: At this stage, we are not generating a specialised QIR - that would first happen if we pass the command line argument `--apply`. The code that is executed is the code that configures components and passes (hence setting the profile up).